### PR TITLE
[DX-1185]: Clean up dependencies

### PR DESCRIPTION
**Pull request by matthewmuscat**
# Summary
This PR removes unnecessary packages, and moves packages from dependencies to devDependencies where necessary.

### For context:
The [ethers package](https://www.npmjs.com/package/ethers) contains all of the [@ethersproject/](https://www.npmjs.com/package/@ethersproject/providers) packages as submodules. This is how we are able to reference such packages such as [@ethersproject/abstract-signer](https://www.npmjs.com/package/@ethersproject/abstract-signer), even though they are not within our `package.json` explicitly. 

We don't need to do this, and is better this way, as [ethers](https://www.npmjs.com/package/ethers) manages [the dependencies of the submodules already. 
](https://github.com/ethers-io/ethers.js/blob/master/packages/abstract-signer/package.json#L4) Doing this ourselves would only cause dependency version conflicts, which would break our build.

# Why the changes
Cleaner bundle

# Things worth calling out

### CODE REVIEW

**Review by zatlite — APPROVED** [review]
Looks good to me. 

Do we want to upgrade some packages to get rid of the dependabot alerts we currently have in `main`? @Kastet

**Comment by Kastet** [comment]
Note `ethereumjs-wallet` will be removed too (in this PR that https://github.com/immutable/imx-core-sdk/pull/150/files)

File: package.json [hunk 1]
```diff
@@ -80,21 +80,17 @@
...
-    "typescript": "^4.6.3"
+    "typescript": "^4.6.3",
+    "dts-bundle-generator": "^6.12.0",
+    "typedoc": "^0.23.10"
   },
   "dependencies": {
-    "@ethersproject/bignumber": "^5.6.2",
-    "@ethersproject/providers": "^5.6.4",
-    "@ethersproject/units": "^5.6.1",
-    "@ethersproject/wallet": "^5.6.2",
```
Comment (Kastet) [inline package.json:89]: question - does `ethers` provide more stuff than we need? Would it be better to remove `ethers` and only target the ones we actually use?

Comment (matthewmuscat) [reply package.json:89]: Yes it does. Probably yes, although the concern is the same as this PR description states where we have to manage the versions of each module and being compatible with each other.

**Review by Kastet — APPROVED** [review]

**Pull request merged**